### PR TITLE
refactor: make annotation getter async and extract from texture loading

### PR DESCRIPTION
**Pull request by haakonflatval-cognite**
#### Type of change

Refactor 

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/REV-906

## Description :pencil:

- Factor out annotations to not go through cache. The hypothesis is that the annotations will not pose a significant memory footprint, even though they are never expelled from memory in a session
- Make annotation getter async
- Outsource default styling handling to each revision

## How has this been tested? :mag:

- Start Reveal examples
- Open https://localhost:3000/?project=3d-test&env=greenfield-3d&modelId=4923307033982497&revisionId=3100284276970593
- Load 360-image set `helideck-site-2-jpeg`
- Enter lower-most 360 image and the one beside it (the latter has one annotation on the ground)
- Observe that annotations are loaded and displayed correctly
- Observe that the annotation default styling (set from the UI) works properly
- Note that clicking an annotation will set a custom appearance overriding the default style, but it should disappear when clicking another annotation

## Checklist :ballot_box_with_check:

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [ ] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [ ] I have refactored the code for readability to the best of my ability.
- [ ] I have checked that my changes do not introduce regressions in the public documentation.
- [ ] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [ ] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [ ] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [ ] I have noted down and am currently tracking any technical debt introduced in this PR.
- [ ] I have thoroughly thought about the architecture of this implementation.
- [x] I have asked peers to test this feature.

### CODE REVIEW

File: viewer/packages/360-images/src/collection/DefaultImage360Collection.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@ export class DefaultImage360Collection implements Image360Collection {
 
   private _needsRedraw: boolean = false;
 
-  private _defaultStyle: Image360AnnotationAppearance = {};
+  private readonly _defaultStyle: { appearance: Image360AnnotationAppearance } = { appearance: {} };
```
Comment (Savokr) [inline viewer/packages/360-images/src/collection/DefaultImage360Collection.ts:40]: Why this change? Is it planned to include more things?

File: viewer/packages/360-images/src/Image360Facade.ts [hunk 2]
```diff
@@ -60,7 +60,14 @@ export class Image360Facade<T> {
...
 
   public preload(entity: Image360Entity, revision: Image360RevisionEntity, lockDownload?: boolean): Promise<void> {
-    return this._image360Cache.cachedPreload(entity, revision, lockDownload);
+    const annotationPromise = revision.getAnnotations();
+    const cacheLoadPromise = this._image360Cache.cachedPreload(entity, revision, lockDownload);
+
+    return awaitPromises();
+
+    async function awaitPromises(): Promise<void> {
+      await Promise.all([annotationPromise, cacheLoadPromise]);
```
Comment (Savokr) [inline viewer/packages/360-images/src/Image360Facade.ts:69]: Do we care about the order of resolving these promises here?

**Review by Savokr — APPROVED** [review]
Looks good! One thing I noticed though is that sometimes 360 image cubemap "flickers" once when clicked for being entered. Doesn't happen every time, though. Not sure this PR introduced it but it is not present in latest reveal version in Data Exploration, so seems like something recent. Here is a video of it: 

https://user-images.githubusercontent.com/45129444/235920321-65793d0b-2af9-49aa-b8b6-030ef915d909.mov

File: viewer/packages/360-images/src/Image360Facade.ts (hunk 2 above)
Comment (haakonflatval-cognite) [reply viewer/packages/360-images/src/Image360Facade.ts:69]: Not really, no... Do you mean I should be using some other function to await them?

File: viewer/packages/360-images/src/collection/DefaultImage360Collection.ts (hunk 1 above)
Comment (haakonflatval-cognite) [reply viewer/packages/360-images/src/collection/DefaultImage360Collection.ts]: No, fixed :+1:

**Comment by haakonflatval-cognite** [comment]
>sometimes 360 image cubemap "flickers" once when clicked for being entered
I think this is because there is a small window between where we set the cubemap visible (line 230 in `Image360ApiHelper` currently) and where it is being incrementally made more opaque while the camera moves towards it. I think the effect appears if we render a frame in that window. Created a bug: https://cognitedata.atlassian.net/browse/REV-934

File: viewer/packages/360-images/src/Image360Facade.ts (hunk 2 above)
Comment (Savokr) [reply viewer/packages/360-images/src/Image360Facade.ts:69]: No, just curious since before this PR the order of things was defined explicitly as far as I understood.
